### PR TITLE
feat: added job Priorities

### PR DESCRIPTION
**Pull request by cl-shaharshavit-rnd**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         |  ✔                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✔                                                                       |

- added priorities support
- fixed lint warnings
- fixed creating new job without tasks field

closes: #20 

Further  information:

### CODE REVIEW

File: src/jobs/models/taskManager.ts [hunk 1]
```diff
@@ -66,6 +67,16 @@ export class TaskManager {
...
   }
 
+  public async retrieveAndStart(req: IRetrieveAndStartRequest): Promise<IGetTaskResponse> {
+    const repo = await this.getRepository();
+    this.logger.log('info', `retrieving and updating task for job type: ${req.jobType} and task type: ${req.taskType}`);
+    const res = await repo.retrieveAndUpdate(req.jobType, req.taskType);
+    if (res === undefined) {
+      throw new EntityNotFound('Pending task was not found');
+    }
+    return res;
```
Comment (asafMasa) [inline src/jobs/models/taskManager.ts:77]: add log of what was retrieved?

File: src/jobs/routes/taskRouter.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { Router } from 'express';
+import { FactoryFunction } from 'tsyringe';
+import { TaskController } from '../controllers/taskController';
+
+const taskRouterFactory: FactoryFunction<Router> = (dependencyContainer) => {
+  const router = Router();
+  const tasksController = dependencyContainer.resolve(TaskController);
+
+  router.post('/:jobType/:taskType/startPending', tasksController.retrieveAndStart);
```
Comment (asafMasa) [inline src/jobs/routes/taskRouter.ts:9]: add integration test

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/jobs/models/taskManager.ts (hunk 1 above)
Comment (cl-shaharshavit-rnd) [reply src/jobs/models/taskManager.ts:78]: added

File: src/jobs/routes/taskRouter.ts (hunk 2 above)
Comment (cl-shaharshavit-rnd) [reply src/jobs/routes/taskRouter.ts:9]: added

**Review by asafMasa — APPROVED** [review]

**Pull request merged**